libsolutil/StringUtils.h

```diff
@@ -168,12 +168,44 @@ inline bool isDigit(char _c)
 	return isdigit(_c, std::locale::classic());
 }
 
-// Checks if character is printable using classic "C" locale
+/// Checks if character is printable using classic "C" locale
 /// @param _c character to be checked
 /// @return true if _c is a printable character, false otherwise.
 inline bool isPrint(char _c)
 {
 	return isprint(_c, std::locale::classic());
 }
 
+/// Adds a prefix to every line in the input.
+/// @see printPrefixed()
+std::string prefixLines(
+	std::string const& _input,
+	std::string const& _prefix,
+	bool _trimPrefix = true
+);
+
+/// Prints to a stream, adding a prefix to every line in the input.
+/// Assumes \n as the line separator.
+/// @param _trimPrefix If true, the function avoids introducing trailing whitespace on empty lines.
+///     This is achieved by removing trailing spaces from the prefix on such lines.
+///     Note that tabs and newlines are not removed, only spaces are.
+/// @param _finalNewline If true, an extra \n will be printed at the end of @a _input if it does
+///     not already end with one.
+void printPrefixed(
+	std::ostream& _output,
+	std::string const& _input,
+	std::string const& _prefix,
+	bool _trimPrefix = true,
+	bool _ensureFinalNewline = true
+);
+
+/// Adds a standard indent of 4 spaces to every line in the input.
+/// Assumes \n as the line separator.
+/// @param _indentEmptyLines If true, the indent will be applied to empty lines as well, resulting
+///     such lines containing trailing whitespace.
+inline std::string indent(std::string const& _input, bool _indentEmptyLines = false)
+{
+	return prefixLines(_input, "    ", !_indentEmptyLines);
```

Comment:
```suggestion
	return prefixLines(_input, std::string(4, ' '), !_indentEmptyLines);
```

I think it's clearer this way, at least with regards to the number of spaces in the indentation.

Reply:
I don't mind either way, but the PR is merged now so I'll leave it as is (the comment was posted while the PR was already on automerge and I did not notice it in time).